File: ios/SwiftKotlination/SwiftKotlinationTests/Sources/Navigation/Mocks/CoordinatorMock.swift
import XCTest
@testable import SwiftKotlination

final class CoordinatorMock: CoordinatorProtocol {
    private(set) var didStart = false
    private(set) var didOpenStory = false
    private(set) var didOpenUrl = false
    
    func start() {
        didStart = true
    }
    
    func open(_ story: Story) {
        didOpenStory = true
    }
    
    func open(_ url: URL) {
        didOpenUrl = true
    }
}

File: android/SwiftKotlination/app/src/test/kotlin/fr/jhandguy/swiftkotlination/coordinator/mocks/CoordinatorMock.kt
package fr.jhandguy.swiftkotlination.coordinator.mocks

import android.content.Intent
import fr.jhandguy.swiftkotlination.coordinator.CoordinatorInterface
import fr.jhandguy.swiftkotlination.features.story.model.Story

class CoordinatorMock : CoordinatorInterface {
    var isStarted = false
    var isStoryOpen = false
    var isUrlOpen = false
    var isFinished = false

    override fun start(): Intent {
        isStarted = true
        return Intent()
    }

    override fun open(story: Story): Intent {
        isStoryOpen = true
        return Intent()
    }

    override fun open(url: String): Intent {
        isUrlOpen = true
        return Intent()
    }

    override fun finish() {
        isFinished = true
    }
}
